instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -731,44 +735,211 @@ def _instrument(self, **kwargs):
+                    span.set_attribute(
+                        ERROR_TYPE, type(exception).__qualname__
+                    )
+                raise exception.with_traceback(exception.__traceback__)
+
+        return response
+
+    @staticmethod
+    async def _handle_async_request_wrapper(  # pylint: disable=too-many-locals
+        wrapped,
+        instance,
+        args,
+        kwargs,
+        tracer,
+        sem_conv_opt_in_mode,
+        async_request_hook,
+        async_response_hook,
+    ):
+        if not is_http_instrumentation_enabled():
+            return await wrapped(*args, **kwargs)
+
+        method, url, headers, stream, extensions = _extract_parameters(
+            args, kwargs
+        )
+        method_original = method.decode()
+        span_name = _get_default_span_name(method_original)
+        span_attributes = {}
+        # apply http client response attributes according to semconv
+        _apply_request_client_attributes_to_span(
+            span_attributes,
+            url,
+            method_original,
+            sem_conv_opt_in_mode,
+        )
+
+        request_info = RequestInfo(method, url, headers, stream, extensions)
+
+        with tracer.start_as_current_span(
+            span_name, kind=SpanKind.CLIENT, attributes=span_attributes
+        ) as span:
+            exception = None
+            if callable(async_request_hook):
+                await async_request_hook(span, request_info)
+
+            _inject_propagation_headers(headers, args, kwargs)
+
+            try:
+                response = await wrapped(*args, **kwargs)
+            except Exception as exc:  # pylint: disable=W0703
+                exception = exc
+                response = getattr(exc, "response", None)
+
+            if isinstance(response, (httpx.Response, tuple)):
+                status_code, headers, stream, extensions, http_version = (
+                    _extract_response(response)
+                )
+
+                if span.is_recording():
+                    # apply http client response attributes according to semconv
+                    _apply_response_client_attributes_to_span(
+                        span,
+                        status_code,
+                        http_version,
+                        sem_conv_opt_in_mode,
+                    )
+
+                if callable(async_response_hook):
+                    await async_response_hook(
+                        span,
+                        request_info,
+                        ResponseInfo(status_code, headers, stream, extensions),
+                    )
+
+            if exception:
+                if span.is_recording() and _report_new(sem_conv_opt_in_mode):
+                    span.set_attribute(
+                        ERROR_TYPE, type(exception).__qualname__
+                    )
+                raise exception.with_traceback(exception.__traceback__)
+
+        return response
+
     def instrument_client(
+        self,
```

Comment:
Why is this no longer a staticmethod? Seems like a significant breaking change that wasn't documented in the release notes

Reply:
I raised this as an issue in #2998 and it was fixed in #3003.